### PR TITLE
[BEAM-11182][Website revamp] Implemented Stay up to date with Beam and Changed Works with components

### PR DESCRIPTION
**Pull request by bntnam**
Home page:
+ Implemented new component "Stay up to date with Beam"
+ Changed "Works with" component

Note: Copies/Texts will be updated later by Gris or the community. For now, it's just the implementation of the design.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by bntnam** [comment]
Review: @TheNeuralBit

File: website/www/site/i18n/home/calendar/en.yaml
Comment (TheNeuralBit) [inline website/www/site/i18n/home/calendar/en.yaml:20]: In the future, will this data be pulled from recent blog posts, or will we have to keep it up to date?

File: website/www/site/i18n/home/calendar/en.yaml
Comment (TheNeuralBit) [inline website/www/site/i18n/home/calendar/en.yaml:42]: What is the difference between files like this (`i18n/**/en.yaml`), and the miscellaneous `en` directories in `data`? AFAICT the difference is just that i18n just has simpler id - translation while the config in data can be more complex and structured.

Based on that distinction it seems like much of this would be more appropriate in `data`

**Review by TheNeuralBit — COMMENTED** [review]
Could you add TODOs around text that's just a placeholder demonstrating the design? That way we won't forgot to swap it out before merging to master

File: website/www/site/i18n/home/calendar/en.yaml
Comment (bntnam) [reply website/www/site/i18n/home/calendar/en.yaml:22]: Great that you asked. Yeah, in the future, we'll have a separate stage for functionality improvement (the current stage is about design implementation). And the event will be synced with the event page (which will be implemented later).

File: website/www/site/i18n/home/calendar/en.yaml
Comment (bntnam) [reply website/www/site/i18n/home/calendar/en.yaml:44]: Yeah. What I understand is, i18n directory is naturally offered by the Hugo plugin with the same name and it's used for localizing strings, syntax  `{{ i18n "id here" }}` or `{{ T "id here" }}`. First, I got stuck when I needed to use a loop structure, plus translation together. Then I figured out that `data` let me to apply a loop structure on it as an array, and i18n helped me to separate directories according to the main language (data/en, data/pl, data/sp, etc.). I know it's a bit annoying to have 2 places, but simply think that one for strings and one for arrays. 😛

**Comment by bntnam** [comment]
Added comments and resolved conflicts!

**Comment by bntnam** [comment]
Added comments and resolved conflicts

File: website/www/site/i18n/home/calendar/en.yaml
Comment (TheNeuralBit) [reply website/www/site/i18n/home/calendar/en.yaml:44]: Gotcha. It seems like with that pattern home-calendar* items could be represented in data with a structure like:
```
home-calendar-first:
- category-tag: "Category"
- category-date: "2020/02/21"
- card-title: "Performance-Driven Runtime Type Checking for the Python SDK"
- card-author: "Saavan Nanavati"
```

File: website/www/site/data/en/calendar_events.yaml
Comment (TheNeuralBit) [inline website/www/site/data/en/calendar_events.yaml:25]: Is it possible for the dates to be represented as some kind of timestamp literal that can be internationalized? For example in the US we usually write dates with the year last - 02/21/2020. Or we may want to have the option to spell out the date in the future.

**Review by TheNeuralBit — APPROVED** [review]
Couple more questions, but they don't need to block merging

**Comment by TheNeuralBit** [comment]
FYI @bntnam I just merged master into website-revamp (bd825f574e342cfa83fb09767c7d5a19a3accc55) to make sure we don't get too far out of sync.

File: website/www/site/data/en/calendar_events.yaml
Comment (bntnam) [reply website/www/site/data/en/calendar_events.yaml:25]: Noted. Thanks for your feedback. Let me tell my PM to add them into Jira.

**Pull request merged**
